src/main/org/codehaus/groovy/util/ManagedConcurrentValueMap.java

```diff
@@ -67,8 +67,8 @@ public void put(final K key, V value) {
         ManagedReference<V> ref = new ManagedReference<V>(bundle, value) {
             @Override
             public void finalizeReference() {
-                super.finalizeReference();
                 internalMap.remove(key, get());
```

Comment:
`get()` here returns V, but internalMap is `ConcurrentHashMap<K,ManagedReference<V>>`, so think it should be:

```
internalMap.remove(key, this);
```